Sources/ObjectEncoder/Encoder.swift

```diff
@@ -175,22 +175,9 @@ private struct _KeyedEncodingContainer<Key: CodingKey> : KeyedEncodingContainerP
     // MARK: - Swift.KeyedEncodingContainerProtocol Methods
 
     var codingPath: [CodingKey] { return encoder.codingPath }
-    func encodeNil(forKey key: Key)               throws { try encoder(for: key).encodeNil() }
-    func encode(_ value: Bool, forKey key: Key)   throws { try encoder(for: key).encode(value) }
-    func encode(_ value: Int, forKey key: Key)    throws { try encoder(for: key).encode(value) }
-    func encode(_ value: Int8, forKey key: Key)   throws { try encoder(for: key).encode(value) }
-    func encode(_ value: Int16, forKey key: Key)  throws { try encoder(for: key).encode(value) }
-    func encode(_ value: Int32, forKey key: Key)  throws { try encoder(for: key).encode(value) }
-    func encode(_ value: Int64, forKey key: Key)  throws { try encoder(for: key).encode(value) }
-    func encode(_ value: UInt, forKey key: Key)   throws { try encoder(for: key).encode(value) }
-    func encode(_ value: UInt8, forKey key: Key)  throws { try encoder(for: key).encode(value) }
-    func encode(_ value: UInt16, forKey key: Key) throws { try encoder(for: key).encode(value) }
-    func encode(_ value: UInt32, forKey key: Key) throws { try encoder(for: key).encode(value) }
-    func encode(_ value: UInt64, forKey key: Key) throws { try encoder(for: key).encode(value) }
-    func encode(_ value: Float, forKey key: Key)  throws { try encoder(for: key).encode(value) }
-    func encode(_ value: Double, forKey key: Key) throws { try encoder(for: key).encode(value) }
-    func encode(_ value: String, forKey key: Key) throws { try encoder(for: key).encode(value) }
-    func encode<T>(_ value: T, forKey key: Key)   throws where T: Encodable { try encoder(for: key).encode(value) }
+    func encodeNil(forKey key: Key)             throws { try encoder(for: key).encodeNil() }
+    func encode<T>(_ value: T, forKey key: Key) throws where T: Primitive { try encoder(for: key).encode(value) }
```

Comment:
Because no one has time to rewrite to that way? 🤷🏻‍♂️
Since it was able to rewrite to a method similar to that in [Yams](https://github.com/jpsim/Yams/blob/master/Sources/Yams/Encoder.swift#L137), it may be rewritten in `JSONEncoder`, too.